### PR TITLE
lsp: Address empty module issues on rename

### PR DESCRIPTION
**Pull request by charlieegan3**
This PR makes changes to where the contents of files come from when being moved. Previously, preference was given to the contents on disk. This turned out to be incorrect as the files on disk are often empty until saved.

I found that I was able to replicate one of the empty module problems by:

- creating a file, and not saving it
- creating a dir
- editing the file
- moving the file into the dir while the file on disk is empty

before

https://github.com/user-attachments/assets/7bbb6442-5dc1-4897-81e2-8f2356d2e7c4

after

https://github.com/user-attachments/assets/c5a5ac35-1867-4214-8ad2-3cb9aa75fcd6

### CODE REVIEW

File: internal/lsp/server.go [hunk 1]
```diff
@@ -747,6 +747,8 @@ func (l *LanguageServer) StartTemplateWorker(ctx context.Context) {
 			newContents, err := l.templateContentsForFile(evt.URI)
 			if err != nil {
 				l.logError(fmt.Errorf("failed to template new file: %w", err))
+
+				continue
```
Comment (charlieegan3) [inline internal/lsp/server.go:751]: This is not the fix, but when there is an error in templating we should skip the rest.

File: internal/lsp/server.go [hunk 2]
```diff
@@ -790,6 +792,14 @@ func (l *LanguageServer) templateContentsForFile(fileURI string) (string, error)
 		return "", errors.New("file already has contents, templating not allowed")
 	}
 
+	diskContent, err := os.ReadFile(uri.ToPath(l.clientIdentifier, fileURI))
```
Comment (charlieegan3) [inline internal/lsp/server.go:795]: this is also not the fix but is an additional guard for templating over files that have contents on disk already.

File: internal/lsp/server.go [hunk 3]
```diff
@@ -1955,17 +1965,24 @@ func (l *LanguageServer) handleWorkspaceDidRenameFiles(
...
 		}
 
-		_, content, err := cache.UpdateCacheForURIFromDisk(
-			l.cache,
-			uri.FromPath(l.clientIdentifier, renameOp.NewURI),
-			uri.ToPath(l.clientIdentifier, renameOp.NewURI),
-		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to update cache for uri %q: %w", renameOp.NewURI, err)
+		content, ok := l.cache.GetFileContents(renameOp.OldURI)
```
Comment (charlieegan3) [inline internal/lsp/server.go:1968]: This is the fix for the empty module issue, were we load from the cache first, and only load from disk if it's missing or empty in the cache

File: internal/lsp/server.go (hunk 2 above)
Comment (anderseknert) [reply internal/lsp/server.go:795]: Any added safeguards are good 👍

File: internal/lsp/server.go (hunk 3 above)
Comment (anderseknert) [reply internal/lsp/server.go:1968]: Nice!

**Review by anderseknert — APPROVED** [review]
LGTM! Test would be good, as with anything touching the LSP, but I know they can be hard to write given the many steps required in sequence. Let's look into how we improve on that later 👍

**Pull request merged**